### PR TITLE
fix(gce-node): disable sshguard on gce ubuntu bionic

### PR DESCRIPTION
**Pull request by bentsi**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [ ] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/cluster_gce.py [hunk 1]
```diff
@@ -300,6 +301,13 @@ def _create_instance(self, node_index, dc_idx, spot=False):
         # Name must start with a lowercase letter followed by up to 63
         # lowercase letters, numbers, or hyphens, and cannot end with a hyphen
         assert len(name) <= 63, "Max length of instance name is 63"
+        startup_script = cluster.Setup.get_startup_script()
+        if "ubuntu-bionic" in self.params.get("scylla_linux_distro", ""):
```
Comment (roydahan) [inline sdcm/cluster_gce.py:305]: Does it cover both Debian 9 and Ubuntu 16,18?

Comment (bentsi) [reply sdcm/cluster_gce.py:305]: does the issue happen on ubuntu 16 and Debian 9?

Comment (bentsi) [reply sdcm/cluster_gce.py:305]: I checket: it happens only on Ubuntu 16, 18

Comment (roydahan) [reply sdcm/cluster_gce.py]: Are you sure??
I’m almost sure it happens on deb9

Comment (roydahan) [reply sdcm/cluster_gce.py]: Are you sure??
I’m almost sure it happens on deb9

Comment (bentsi) [reply sdcm/cluster_gce.py]: `sshguard` is not installed on Debian 9
```
Apr 27 13:09:59 rolling-upgrade-3-3-debian-stretch-db-node-d99b76c4-0-2 sshd[2575]: pam_unix(sshd:session): session closed for user scylla-test
Apr 27 13:10:04 rolling-upgrade-3-3-debian-stretch-db-node-d99b76c4-0-2 sshd[2585]: Accepted publickey for scylla-test from 10.142.0.5 port 52424 ssh2: RSA SHA256:ORclW5uOPUGIzX/HqWCBo6YH8GIz8D8MPjTEd0njNms
Apr 27 13:10:04 rolling-upgrade-3-3-debian-stretch-db-node-d99b76c4-0-2 sshd[2585]: pam_unix(sshd:session): session opened for user scylla-test by (uid=0)
Apr 27 13:10:04 rolling-upgrade-3-3-debian-stretch-db-node-d99b76c4-0-2 sshd[2585]: pam_unix(sshd:session): session closed for user scylla-test
/sshguard
Pattern not found
root@rolling-upgrade-3-3-debian-stretch-db-node-d99b76c4-0-2:/home/scylla-test# cat /etc/os
osconfig/   os-release  
root@rolling-upgrade-3-3-debian-stretch-db-node-d99b76c4-0-2:/home/scylla-test# cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
VERSION_CODENAME=stretch
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
root@rolling-upgrade-3-3-debian-stretch-db-node-d99b76c4-0-2:/home/scylla-test# systemctl status sshguard
Unit sshguard.service could not be found.
root@rolling-upgrade-3-3-debian-stretch-db-node-d99b76c4-0-2:/home/scylla-test# 

```

**Comment by bentsi** [comment]
checked on https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/bentsi-staging/job/bentsi-artifacts-ubuntu1804/3/
merging this

File: sdcm/cluster_gce.py [hunk 2]
```diff
@@ -300,6 +301,13 @@ def _create_instance(self, node_index, dc_idx, spot=False):
         # Name must start with a lowercase letter followed by up to 63
         # lowercase letters, numbers, or hyphens, and cannot end with a hyphen
         assert len(name) <= 63, "Max length of instance name is 63"
+        startup_script = cluster.Setup.get_startup_script()
+        if self.params.get("scylla_linux_distro", "") in ("ubuntu-bionic", "ubuntu-xenial"):
+            # we need to disable sshguard to prevent blocking connections from the builder
+            startup_script += dedent("""
```
Comment (amoskong) [inline sdcm/cluster_gce.py:307]: Can we disable sshguard as early as possible?
Currently it's after restarting sshd.

**Comment by amoskong** [comment]
@bentsi We need this PR for 3.3 and 2019.1

**Comment by bentsi** [comment]
can't backport to 2019.1, too many features missing. If it is needed please create a backport PR.

**Comment by amoskong** [comment]
> can't backport to 2019.1, too many features missing. If it is needed please create a backport PR.

2019.1 isn't urgent, I will create a PR for backport.
Thanks.

**Comment by amoskong** [comment]
> > can't backport to 2019.1, too many features missing. If it is needed please create a backport PR.
> 
> 2019.1 isn't urgent, I will create a PR for backport.
> Thanks.

Just sent a PR for 2019.1 backport: https://github.com/scylladb/scylla-cluster-tests/pull/2149

**Pull request merged**